clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -161,7 +162,9 @@
         new NotControllerException("This is not the correct controller for this cluster.")),
     INVALID_REQUEST(42,
         new InvalidRequestException("This most likely occurs because of a request being malformed by the client library or" +
-            " the message was sent to an incompatible broker. See the broker logs for more details."));
+            " the message was sent to an incompatible broker. See the broker logs for more details.")),
+    MESSAGE_FORMAT_TOO_OLD(43,
```

Comment:
I was thinking about that name as well, but it is a little confusing because the message format is actually valid, it is just the function is not supported. Also we already have InvalidMessageException. Message format too old might be a generic enough exception, e.g. KIP-82 will probably introduce some new functions into the messages and may not be supported by the old message format.